commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SectionExt.kt

```diff
@@ -1,10 +1,32 @@
 package ebi.ac.uk.model.extensions
 
+import ebi.ac.uk.model.ExtendedSection
+import ebi.ac.uk.model.File
 import ebi.ac.uk.model.Section
+import ebi.ac.uk.model.constants.SectionFields
+import ebi.ac.uk.util.collections.ifLeft
 
-fun Section.allFiles() = files.map { it.fold({ file -> listOf(file) }, { table -> table.elements }) }.flatten()
+var Section.libraryFile: String?
+    get() = this[SectionFields.LIB_FILE]
+    set(value) {
+        value?.let { this[SectionFields.LIB_FILE] = it }
+    }
 
-fun Section.allReferencedFiles() = libraryFile?.referencedFiles?.toList() ?: listOf()
+fun Section.allFiles() = files.map { it.fold({ file -> listOf(file) }, { table -> table.elements }) }.flatten()
 
 fun Section.allSections() =
     sections.map { it.fold({ section -> listOf(section) }, { table -> table.elements }) }.flatten()
+
+fun ExtendedSection.allReferencedFiles(): List<File> {
```

Comment:
I think is worth to seprate `SectionExt` from `ExtendedSectionExt`

Reply:
Done